protocol/contracts/amo/Ramos.sol

```diff
@@ -119,53 +156,89 @@ contract RAMOS is Ownable, Pausable {
+     * @param quoteTokenMaxAmount Maximum `quoteToken` amount per rebalance
+     * @param protocolTokenMaxAmount Maximum protocolToken amount per rebalance
      */
-    function setMaxRebalanceAmounts(uint256 bptMaxAmount, uint256 stableMaxAmount, uint256 templeMaxAmount) external onlyOwner {
-        if (bptMaxAmount == 0 || stableMaxAmount == 0 || templeMaxAmount == 0) {
-            revert AMOCommon.InvalidMaxAmounts(bptMaxAmount, stableMaxAmount, templeMaxAmount);
+    function setMaxRebalanceAmounts(uint256 bptMaxAmount, uint256 quoteTokenMaxAmount, uint256 protocolTokenMaxAmount) external onlyElevatedAccess {
+        if (bptMaxAmount == 0 || quoteTokenMaxAmount == 0 || protocolTokenMaxAmount == 0) {
+            revert AMOCommon.InvalidMaxAmounts(bptMaxAmount, quoteTokenMaxAmount, protocolTokenMaxAmount);
         }
         maxRebalanceAmounts.bpt = bptMaxAmount;
-        maxRebalanceAmounts.stable = stableMaxAmount;
-        maxRebalanceAmounts.temple = templeMaxAmount;
-        emit SetMaxRebalanceAmounts(bptMaxAmount, stableMaxAmount, templeMaxAmount);
+        maxRebalanceAmounts.quoteToken = quoteTokenMaxAmount;
+        maxRebalanceAmounts.protocolToken = protocolTokenMaxAmount;
+        emit SetMaxRebalanceAmounts(bptMaxAmount, quoteTokenMaxAmount, protocolTokenMaxAmount);
     }
 
-    // @notice percentage bounds (in bps) beyond which to rebalance up or down
-    function setRebalancePercentageBounds(uint64 belowTPF, uint64 aboveTPF) external onlyOwner {
-        if (belowTPF > BPS_PRECISION || aboveTPF > BPS_PRECISION) {
-            revert AMOCommon.InvalidBPSValue(belowTPF);
+    /// @notice Set maximum percentage bounds (in bps) beyond which to rebalance up or down
+    function setRebalancePercentageBounds(uint64 belowTpi, uint64 aboveTpi) external onlyElevatedAccess {
+        if (belowTpi > BPS_PRECISION || aboveTpi > BPS_PRECISION) {
+            revert AMOCommon.InvalidBPSValue(belowTpi);
         }
-        rebalancePercentageBoundLow = belowTPF;
-        rebalancePercentageBoundUp = aboveTPF;
+        rebalancePercentageBoundLow = belowTpi;
+        rebalancePercentageBoundUp = aboveTpi;
 
-        emit SetRebalancePercentageBounds(belowTPF, aboveTPF);
+        emit SetRebalancePercentageBounds(belowTpi, aboveTpi);
     }
 
-    function setTemplePriceFloorNumerator(uint128 _numerator) external onlyOwner {
-        templePriceFloorNumerator = _numerator;
+    /**
+     * @notice Set the Treasury Price Index (TPI) Oracle
+     */
+    function setTpiOracle(address newTpiOracle) external override onlyElevatedAccess {
+        emit TpiOracleSet(newTpiOracle);
+        tpiOracle = ITreasuryPriceIndexOracle(newTpiOracle);
+    }
 
-        emit SetTemplePriceFloorNumerator(_numerator);
+    /**
+     * @notice Set the Treasury Price Index (TPI) Oracle
+     */
+    function setProtocolTokenVault(address vault) external override onlyElevatedAccess {
+        emit ProtocolTokenVaultSet(vault);
+
```

Comment:
Is this extra check warranted for the bytecode increase? It will only be called rarely by elevated access, and it wouldn't be the case that the new vault is the same as the old one

Reply:
Decided not to implement